org.jacoco.core/src/org/jacoco/core/runtime/InjectedClassRuntime.java

```diff
@@ -48,6 +56,71 @@ public InjectedClassRuntime(final Class<?> locator,
 				'/') + '/' + simpleClassName;
 	}
 
+	/**
+	 * Creates a new {@link InjectedClassRuntime} that will define class in the
+	 * {@code java.lang} package. To do so this method opens package
+	 * {@code java.lang} to the unnamed module of a new ClassLoader that is used
+	 * to create an instance of {@link InjectedClassRuntime}, so this <a href=
+	 * "https://docs.oracle.com/javase/specs/jvms/se11/html/jvms-5.html#jvms-5.3.6">
+	 * unnamed module is distinct from all run-time modules (including unnamed
+	 * modules) bound to other class loaders</a>.
+	 *
+	 * @param instrumentation
+	 *            instrumentation interface
+	 * @return new runtime instance, null if JVM version is lower than 9
+	 * @throws Exception
+	 *             if unable to create
+	 */
+	public static IRuntime createFor(final Instrumentation instrumentation)
+			throws Exception {
+		try {
+			Class.forName("java.lang.Module");
+		} catch (final ClassNotFoundException e) {
+			return null;
+		}
+		final ClassLoader classLoader = new ClassLoader() {
+			@Override
+			protected Class<?> loadClass(String name, boolean resolve)
+					throws ClassNotFoundException {
+				if (!name.equals(InjectedClassRuntime.class.getName())
+						&& !name.equals(Lookup.class.getName())) {
+					return super.loadClass(name, resolve);
+				}
+				final InputStream resourceAsStream = getResourceAsStream(
+						name.replace('.', '/') + ".class");
+				final byte[] bytes;
+				try {
+					bytes = InputStreams.readFully(resourceAsStream);
+				} catch (IOException e) {
```

Comment:
```suggestion
				} catch (final IOException e) {
```
We currently make all locals final, Eclipse will add this if I save the file.